packages/tx/src/types.ts

```diff
@@ -126,14 +145,31 @@ export interface AccessListEIP2930TxData extends TxData {
    * The access list which contains the addresses/storage slots which the transaction wishes to access
    */
   accessList?: AccessListBuffer | AccessList
+}
 
+/**
+ * Fee marked EIP1559 tx data.
+ */
+export interface FeeMarketEIP1559TxData extends AccessListEIP2930TxData {
   /**
-   * The transaction type
+   * The transaction's gas price.
    */
-
-  type?: BNLike
+  gasPrice?: never
+  /**
+   * The maximum inclusion fee per gas (this fee is given to the miner)
+   */
+  maxInclusionFeePerGas?: BNLike
+  /**
+   * The maximum total fee
+   */
+  maxFeePerGas?: BNLike
 }
 
+/**
+ * Buffer values array for EIP2930 transaction
```

Comment:
not just for EIP2930, right?

```suggestion
 * Buffer values array for transactions
```

Reply:
Ah, forgot to update, thanks, this is actually *only* for the legacy txs - since we still have got this situation that we have all things "legacy" just named "Transaction" or "Tx" and I wanted to remain consistent. So even more important to comment correctly. Will update.